why-django.qmd

```diff
@@ -54,99 +54,80 @@ and drawbacks for each option.
-logic and allows developers to work with database models in a Pythonic
-way. It provides good support for migrations, which makes it easy to
-evolve the database schema over time. Django also provides a built-in
-admin interface that allows non-technical users to manage data.
-
-Django provides good support for security through its built-in
-authentication system and its support for common security practices such
-as Cross-Site Request Forgery (CSRF) protection and password hashing. It
-also provides good support for role-based access control and
-fine-grained permissions.
-
-Django provides good support for user management through its built-in
-authentication system and its support for user profiles and social
-authentication. It also provides good support for password resets and
-account activation.
+[Django](https://www.djangoproject.com) is a full-stack framework for building server-rendered web applications following a Model-View-Template pattern.
 
 
 ::: {.columns}
 ::: {.column}
 #### Benefits
 
-- Item 1
+- Has excellent support for API development, including frameworks for REST and GraphQL architectures.
+- With the [REST framework](https://www.django-rest-framework.org/) it offers built-in solutions for all common web development tasks, such as HTTP request handling, serialisation, URL routing, pagination, filtering and internationalisation. 
+- Comes with a wide range of tools for user management, such as a customisable authentication system with support for social authentication and user profiles, and account-activation and password-reset mechanisms.
+- Promotes security best-practices through the built-in authentication system, role-based access control, fine-grained permissions, Cross-Site Request Forgery (CSRF) protection and password hashing.
+- Provides a powerful Object-Relational Mapper (ORM) that abstracts away much of the database logic and allows developers to work with database models in a Pythonic way.
+- Supports all major databases and has a migration mechanism, making it easy to evolve the database schema over time.
+- Has good file storage support, making it easier to work with raw data and image files.
+- Comes with an admin interface out of the box, allowing non-technical users to manage data.
+- Offers built-in form handling capabilities with customisable form classes and form validation logic.
+- Uses server-side rendering by default, enabling fast webpage renders. 
+- Has a template engine for creating dynamic HTML pages.
+- Makes it easy to compose Django applications and has a wide array of well-maintained packages.
+- Has excellent documentation and a large and active developer community.
 :::
 ::: {.column}
 #### Drawbacks
 
-- Item 1
+- Has a steeper learning curve because of its extensive set of features.
+- May be less performant in certain scenarios than the other frameworks considered due to the size of the core framework and its limited support for asynchronous programming.
```

Comment:
I kind of think that the actual answer to this is that you can contrive scenarios where Django is outperformed by e.g. FastAPI (e.g. [this benchmark](https://blog.stackademic.com/speed-matters-a-comparative-performance-analysis-of-fastapi-and-django-2d78fd23f909)), but it's hard to say if these tests are representative of real-world use cases and, at any rate, the difference seems to be marginal.

I guess if you had some sort of safety-critical system that handled lots of concurrent, time-consuming incoming and outgoing requests in real time, then you would really care about each ms gained by eliminating idle waiting periods. But then you may not be wanting to develop your safety-critical system as a Python web app…

I initially put this in because every second blog post claims it in some shape or form, but it would probably be better to change the angle a bit and say something like “Django may require more complex configuration and additional extensions to offer the same level of support for concurrency/async programming as the default offered by FastAPI”.